flow-storm/flow-storm-debugger
# Feat: Wrap Lines in Args and Ret Panes, Add Print Wrap Checkbox

Hey @jpmonettas  ! I'm a Computer Science undergraduate at UFCG with a background in research and hands-on projects around the Clojure programming language. This project caught my attention due to its relevance to my current studies and aligns perfectly with my growing interest and involvement in Clojure-related research.

This pull request addresses [Issue #102](https://github.com/flow-storm/flow-storm-debugger/issues/102) by implementing the requested feature of wrapping lines in Args and Ret panes. The implementation includes the addition of a checkbox, named "print-wrap", for controlling the line wrapping behavior.

## Changes Made
- Added a checkbox named `print-wrap` to control line wrapping in Args and Ret panes.
- Implemented line wrapping using setWrapText for the TextArea containing arguments and return values.
- Added two test expressions in the test file `dev-tester.clj` to demonstrate the functionality: `lorem-ipsum [arg1 arg2 arg3]` and `generate-lorem-ipsum []`.
- Recorded a demonstration video showcasing the new feature, which is included in this pull request.
- Updated the CHANGELOG.md file to reflect the new feature.

## How to Test
1. Clone this branch locally.
2. Run the provided test expressions in the debugger.
3. Toggle the `print-wrap` checkbox to observe the line wrapping behavior.

## Demo Video

https://github.com/flow-storm/flow-storm-debugger/assets/41126948/b06face1-7b2b-4a66-b53b-b13b40044643


## Screenshots

- ![ret-wrap-lines](https://github.com/flow-storm/flow-storm-debugger/assets/41126948/b79920ad-cae8-4d1a-abe2-236d151478ee)

- ![args-wrap-lines](https://github.com/flow-storm/flow-storm-debugger/assets/41126948/abe1e2a8-343b-4daa-9a59-9802fadf0ade)


## Closing Notes
Thank you for considering this pull request. I hope the implemented feature aligns with the project's goals and enhances the user experience. Please feel free to provide feedback or request further changes. I'm eagerly anticipating the opportunity to make further contributions to the project in the near future.
